### ISSUE TITLE
pfSense appliance is currently pretty useless

### ISSUE THREAD

**Issue by ehlers** [GNS3/gns3-registry#394]
In the current state the pfSense appliance doesn't work.

- I haven't found any official location for downloading older releases. Only the latest one is available at <https://www.pfsense.org/download/>. Therefore only the latest version should be in the appliance. And it has to be updated very often, otherwise the user runs into the issue, that the downloadable version is not in the appliance.
- The NanoBSD version is no longer available, only Memstick or ISO installer images.
- The v2.4.3 memstick image from the appliance will very likely use the VGA console, but telnet is used as the console_type. I can't check as this version is no longer downloadable. For the current v2.4.4 you must download the serial memstick image, then the telnet console will work
- The memstick image will boot, but is not directly usable (as l least I found no way to use it). It must be installed in some free disk space. So either you have to extend the memdisk image to add room for a pfSense installation or you must add an empty second hard disk for installation (and remove the memdisk image after installation).

Anyway the loadable pfSense images are **installers**, it needs some extra work to build an image from them. That doesn't fit into the appliance schema, so I would delete it. That doesn't mean, that GNS3 shouldn't continue to support pfSense. But for the installation the user needs a documentation showing all the steps to build an image.

**Comment by ehlers** [comment]
Here my pfSense installation documentation: <https://www.bernhard-ehlers.de/blog/2019/01/08/gns3-create-pfsense-vm.html>

If you like it, you can freely use it (or parts of it) in a GNS3 document for installing pfSense.

**Comment by adosztal** [comment]
Thanks, I'll update the appliance file.

**Comment by josephillips85** [comment]
Hi @ehlers  
i upgrade the template for pfSense 2.4.4 Please if you can please try it.

#404

**Comment by ehlers** [comment]
@josephillips85 Your updated template works very good, some comments though:

It uses the graphical (VNC) console. For users with non-US keyboards that's a pain, the keys create different characters than expected. And no, setting the keyboard type during the installation will not help. But the big advantage is, that the ISO installer is even easier to use than the memstick installer, I used in my documentation. And it's very easy to change the console to serial in the web configurator (advanced system configuration). So from my point of view, it's fine.

Another disadvantage is, that the installation has to be done for each device you use. But as the pfsense installation is very fast and easy, it's not a big deal. And you have no possibility with a GNS3 appliance, to do this installation only once. So it's fine as well.

What can be improved, is the addition of some usage instructions:
- Remind the user to uncompress the ISO
- Installation instructions (just press \<Enter\> to all questions)
- Default credentials for the web interface (admin/pfsense)

To summarize, you've made a very good appliance, adding some usage instructions will make it even better.

**Comment by josephillips85** [comment]
@ehlers Thanks for the recommendations. 

I perform a change to speed up the installation and the boot process replacing the IDE interface to virtio. 
Also add the default credentials and the install instructions.

I don't think we have to remember to the user to decompress the file , because on the upload dialog box is requesting a .iso and not a .gz.

Thanks

**Comment by ehlers** [comment]
Nice, thanks for all your work.

I'll close this issue as soon as your PR gets merged.

**Comment by APoniatowski** [comment]
its outdated again... its on p3 now...
